### ISSUE TITLE
feat: implement the library headers to connect all the projects.

### ISSUE THREAD

**Issue by TahimiLeonBravo** [90milesbridge/react-rainbow#373]
normal https://zpl.io/b63YJ4d
when hover https://zpl.io/2pnxWEy
when active https://zpl.io/2GoMnP3

Requirements:
1. the menuItem should be clickeable
2. the menu should be to close when clicking outside
3. should be keyboard navigation
4. the menu should be to close when press ESC
5. Each option should navigate to the library
